core/src/main/scala/org/bykn/bosatsu/rankn/Type.scala

```diff
@@ -298,17 +297,25 @@ object Type {
 
   case class Meta(id: Long, ref: Ref[Option[Type]])
 
-  def metaTvs(s: List[Type]): Set[Meta] = {
+  object Meta {
+    implicit val orderingMeta: Ordering[Meta] =
+      Ordering.by { m: Meta => m.id }
+  }
+
+  /**
+   * Final the set of all of Metas inside the list of given types
+   */
+  def metaTvs(s: List[Type]): SortedSet[Meta] = {
```

Comment:
the order of these variables is used when we create the forAll. Using a sorted set makes sure that is repeatable. We should really only use a Set when it is used a membership check, and never enumerated.